### PR TITLE
Add compatibility for Kodi 19 Python API

### PR DESCRIPTION
**Pull request by mediaminister**
This pull request adds compatibility with Kodi 19 Python API while preserving backwards compatibility with earlier Kodi versions.

### CODE REVIEW

File: test/xbmcgui.py [hunk 1]
```diff
@@ -110,38 +112,41 @@ class DialogProgress:
...
-    def create(self, heading, line1, line2=None, line3=None):
+    def create(self, heading, message='', line1='', line2='', line3=''):
         """A stub implementation for the xbmcgui DialogProgress class create() method"""
-        self.percentage = 0
+        self.percent = 0
         heading = kodi_to_ansi(heading)
+        message = kodi_to_ansi(message)
         line1 = kodi_to_ansi(line1)
-        print('\033[37;44;1mPROGRESS:\033[35;49;1m [%s] \033[37;1m%s\033[39;0m' % (heading, line1))
+        message = kodi_to_ansi(message)
```
Comment (horstle) [inline test/xbmcgui.py]: Why did you add this twice?

File: lib/inputstreamhelper/kodiutils.py [hunk 2]
```diff
@@ -106,7 +119,9 @@ def yesno_dialog(heading='', message='', nolabel=None, yeslabel=None, autoclose=
     from xbmcgui import Dialog
     if not heading:
         heading = ADDON.getAddonInfo('name')
-    return Dialog().yesno(heading=heading, line1=message, nolabel=nolabel, yeslabel=yeslabel, autoclose=autoclose)
+    if kodi_version_major() < 19:
+        return Dialog().yesno(heading=heading, line1=message, nolabel=nolabel, yeslabel=yeslabel, autoclose=autoclose)
```
Comment (dagwieers) [inline lib/inputstreamhelper/kodiutils.py]: We should probably move this to using kodiutils as well.

File: test/xbmcgui.py (hunk 1 above)
Comment (mediaminister) [reply test/xbmcgui.py]: That is a typo. Thanks for noticing.

File: lib/inputstreamhelper/kodiutils.py (hunk 2 above)
Comment (mediaminister) [reply lib/inputstreamhelper/kodiutils.py]: I don't understand, what do you mean exactly?

Comment (dagwieers) [reply lib/inputstreamhelper/kodiutils.py]: The kodiutils library is an abstraction layer to all Kodi-related calls, and here we aren't using it.

Comment (mediaminister) [reply lib/inputstreamhelper/kodiutils.py]: I'm confused, this code is in **kodiutils.py**

Comment (horstle) [reply lib/inputstreamhelper/kodiutils.py]: @dagwieers You probably didn't see this is inside kodiutils?

Comment (dagwieers) [reply lib/inputstreamhelper/kodiutils.py]: Ok, I was confused. I thought we moved everything to kodiutils in the past, so I was surprised to see this :smile: Move along, nothing to see here...

**Comment by dagwieers** [comment]
@mediaminister Since this has been confirmed to be working, and this is conflicting with my GitHub CI work, I am going to merge this now. If there are additional fixes needed later, we can still add them later. Sorry for the inconvenience.

**Comment by dagwieers** [comment]
I was wondering what the overhead was of using `inspect.getargspec()` instead of a version check.

It would support both older Kodi v19 releases as well as newer Kodi v19 releases (during the transition period). Some users do run e.g. LibreELEC test releases that have older Kodi versions, and are not easily updated. (See #338)
